KB-11624 | BE | DEV | API for Creating consent acknowledgement and reading the consent details.

tarentomaheshvakkund opened the pull request:
1. Created two new APIS - consent acknowledgement and consent read.
2. Implemented cassandra insertrecord method based on the primarykey and composite key.
3. Implemented the test cases.

tarentomaheshvakkund committed:
KB-11624 | BE | DEV | API for Creating consent acknowledgement and reading the consent details.

1. Created two new APIS - consent acknowledgement and consent read.
2. Implemented cassandra insertrecord method based on the primarykey and composite key.
3. Implemented the test cases.

tarentomaheshvakkund committed:
Updated the response structure

tarentomaheshvakkund committed:
Test cases updated.

karthik-tarento commented on src/main/java/com/igot/cb/cassandra/CassandraOperationImpl.java:
```diff
@@ -199,4 +194,50 @@ public void deleteRecord(String keyspaceName, String tableName, Map<String, Obje
+            if (MapUtils.isNotEmpty(otherFields)) {
+                request.putAll(otherFields);
+            }
+            String query = CassandraUtil.getPreparedStatement(keyspaceName, tableName, request);
```
should use simple statement -- not prepared statement.

tarentomaheshvakkund replied:
Replaced with :
 SimpleStatement simpleStatement = SimpleStatement.builder(query)
                    .addPositionalValues(request.values())
                    .build();
            session.execute(simpleStatement);

karthik-tarento commented on src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeController.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.Map;
+
+@RestController
+@RequestMapping("/consent")
```
provide the version details... 
/consent/v1/

tarentomaheshvakkund replied:
Updated with : @RequestMapping("/v1/consent")

karthik-tarento commented on src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeServiceImpl.java:
```diff
@@ -0,0 +1,155 @@
+        declarationDataMap.put(Constants.ADDITIONAL_ATTRIBUTES, additionalDataStr);
+        declarationDataMap.put(Constants.SUBMITTED_AT, submittedAt.format(formatter));
+        try {
+            cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_DECLARATION_ACKNOWLEDGMENT, Constants.CONTENT_ID, contentId, compositeKeyMap, declarationDataMap);
```
when there is an exception -- insertRecord doesn't re-throw.
either you need to re-throw the exception or validate the return response.

tarentomaheshvakkund replied:
Validated the return response:

response = (ApiResponse) cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_DECLARATION_ACKNOWLEDGMENT, Constants.CONTENT_ID, contentId, compositeKeyMap, declarationDataMap);
            if (Constants.FAILED.equals(response.get(Constants.RESPONSE))) {
                logger.error("Error while inserting declaration acknowledgment record in DB: {}",
                        response.get(Constants.ERROR_MESSAGE));
                ProjectUtil.errorResponse(
                        response,
                        "Failed to acknowledge declaration. Please try again later.",
                        HttpStatus.INTERNAL_SERVER_ERROR
                );
                return response;
            }

karthik-tarento commented on src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeController.java:
```diff
@@ -0,0 +1,36 @@
+        this.acknowledgeService = acknowledgeService;
+    }
+
+    @PostMapping("/acknowledge")
```
Need a GET version of this API

tarentomaheshvakkund replied:
Added new API: @GetMapping("/acknowledge/read/{contentId}/{consentId}")

karthik-tarento commented on src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeController.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+
+    @GetMapping("/details/{consentId}")
```
this is not required right now... 
we won't be able to support - if we have more than one consent

tarentomaheshvakkund committed:
Code review comments fixed.

1. Prepared Statement is being replaced with the SImpleStatement.
2. Version of the Controller mappings.
3. When there is an exception validate the return response.
4. Updated the test cases.

tarentomaheshvakkund committed:
Consent Acknowledge read API added.

1. added new API for consent acknowledge read API.
2. Added test cases.
